packages/cashier/src/containers/routes/routes.tsx

```diff
@@ -1,30 +1,17 @@
+import { observer, useStore } from '@deriv/stores';
 import React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router';
-import { connect } from 'Stores/connect';
-import { TClientStore, TCommonStore, TRootStore } from 'Types';
+import { withRouter } from 'react-router';
 import BinaryRoutes from './binary-routes';
 import ErrorComponent from './error-component';
 
-type TRoutesProps = RouteComponentProps & {
-    error: TCommonStore['error'];
-    has_error: TCommonStore['has_error'];
-    is_logged_in: TClientStore['is_logged_in'];
-    is_logging_in: TClientStore['is_logging_in'];
-};
+const Routes = observer(() => {
+    const { client, common } = useStore();
+    const { is_logged_in, is_logging_in } = client;
+    const { error, has_error } = common;
 
-const Routes = ({ error, has_error, is_logged_in, is_logging_in }: TRoutesProps) => {
-    if (has_error) {
-        return <ErrorComponent {...error} />;
-    }
+    if (has_error) return <ErrorComponent {...error} />;
 
     return <BinaryRoutes is_logged_in={is_logged_in} is_logging_in={is_logging_in} />;
-};
+});
 
-// need to wrap withRouter around connect
-// to prevent updates on <BinaryRoutes /> from being blocked
```

Comment:
Not sure if this comment is still valid or not, If it is then I guess we need to wrap `BinaryRoutes` with `observer` but I don’t see any use of stores data in `BinaryRoutes` 🤔